includes/Extension/SearchReplace/class-replace.php

```diff
@@ -85,12 +96,163 @@ public function add_replace_button() {
+				$button = array(
+					'message' => esc_attr__( 'Create a Staging Site', 'string-locator' ),
+					'url'     => $instawp_connect,
+					'classes' => array( 'string-locator-instawp-button', 'disabled' ),
+				);
+
+				if ( '' !== $activated ) {
+					$button['message'] = esc_attr( $activated );
+
+					$button['target'] = 'target="_blank"';
+				}
+				$button['logo-img'] = '<span class="btn-logo-inline"><img src="'. plugins_url( 'views/assets/instawp-logo.svg', STRING_LOCATOR_PLUGIN_FILE ).'" alt="InstaWP logo"></span>';
+
+			} elseif ( self::is_plugin_installed( $plugin_slug ) ) {
+				$url = self::is_plugin_installed( $plugin_slug );
+
+				// The plugin exists but isn't activated yet.
+				$button = array(
+					'message' => esc_attr__( 'Create a Staging Site', 'string-locator' ),
+					// 'url'     => $url,
+					'url'     => "javascript:void 0;",
+					'classes' => array( 'instawp-activate-now' ),
+				);
+
+				$button['logo-img'] = '<span class="btn-logo-inline"><img src="'. plugins_url( 'views/assets/instawp-logo.svg', STRING_LOCATOR_PLUGIN_FILE ).'" alt="InstaWP logo"></span>';
+				if ( '' !== $activate ) {
+					$button['message'] = esc_attr( $activate );
+				}
+			} else {
+				// The plugin doesn't exist.
+				$url    = wp_nonce_url(
+					add_query_arg(
+						array(
+							'action' => 'install-plugin',
+							'plugin' => $plugin_slug,
+						),
+						self_admin_url( 'update.php' )
+					),
+					'install-plugin_' . $plugin_slug
+				);
+				$button = array(
+					'message' => esc_attr__( 'Create a Staging Site', 'string-locator' ),
+					'url'     => $url,
+					'classes' => array( 'sl-instawp-install-now', 'install-now', 'install-' . $plugin_slug ),
+				);
+
+				if ( '' !== $install ) {
+					$button['message'] = esc_attr( $install );
+				}
+
+				$button['logo-img'] = '<span class="btn-logo-inline"><img src="'. plugins_url( 'views/assets/instawp-logo.svg', STRING_LOCATOR_PLUGIN_FILE ).'" alt="InstaWP logo"></span>';
+			}
+
+			if ( ! empty( $classes ) ) {
+				$button['classes'] = array_merge( $button['classes'], $classes );
+			}
+
+			$button['classes'] = implode( ' ', $button['classes'] );
+
+			?>
+			<span class="plugin-card-<?php echo esc_attr( $plugin_slug ); ?>" style="float: right; margin-top: 7px;">
+				<?php echo !empty($button["logo-img"]) ? $button["logo-img"] : '';?>
+				<a href="<?php echo esc_url( $button['url'] ); ?>" class="<?php echo esc_attr( $button['classes'] ); ?>" data-originaltext="<?php echo esc_attr( $button['message'] ); ?>" data-name="<?php echo esc_attr( $plugin_name ); ?>" data-slug="<?php echo esc_attr( $plugin_slug ); ?>" aria-label="<?php echo esc_attr( $button['message'] ); ?>" <?php echo !empty($button["target"]) ? $button["target"] : '';?>> 
+					<?php echo esc_html( $button['message'] ); ?> 
+				</a>
+			</span>
+			<?php
+		}
+
+	}
+
+	/**
+	 * Handle Ajax call to activate the function
+	 *
+	 */
+	public function install_activate_plugin_callback()
```

Comment:
This function handles an admin-ajax callback, but has no nonce verification or user authentication checks.

It also accepts a generic plugin slug as a post value, which may open up the plugin to allow any actor to install any plugin on a site with this plugin active.

This should be replaced with a REST endpoint like the other interactive elements, and also have the `install_plugins` capability check as part of that endpoints `permission_callback` check.